applications/GeoMechanicsApplication/custom_elements/U_Pw_base_element.cpp

```diff
@@ -167,40 +162,31 @@ void UPwBaseElement<TDim, TNumNodes>::Initialize(const ProcessInfo& rCurrentProc
     KRATOS_CATCH("")
 }
 
-//----------------------------------------------------------------------------------------
-template <unsigned int TDim, unsigned int TNumNodes>
-void UPwBaseElement<TDim, TNumNodes>::ResetConstitutiveLaw()
+void UPwBaseElement::ResetConstitutiveLaw()
 {
     KRATOS_TRY
 
-    // erasing stress vectors
-    for (unsigned int i = 0; i < mStressVector.size(); ++i) {
-        mStressVector[i].clear();
-    }
+    std::for_each(mStressVector.begin(), mStressVector.end(),
+                  [](auto& stress_vector) { stress_vector.clear(); });
```

Comment:
It's nice this is no longer a raw for-loop, but you could also achieve the same with a ranged based for-loop (since we're looping over the entire vector anyway), which I always think is slightly easier to read (and write). Let me know what you think!
```suggestion
    for (auto& rStressVector : mStressVectors) {
        rStressVector.clear()
    }
```
This applies to quite some other places as well, I'm not sure what @avdg81's opinion is on this? Might be nice if we're consistent (although it's not a must in my opinion in this case)

Reply:
I agree that for consistency it might be better still to use a range-based `for` loop, rather than `std::for_each`. That STL algorithm was really nice before the range-based `for` loop was available (i.e. prior to C++11). I believe this algorithm has become obsolete now, since it doesn't really add to what you can do with a range-based `for` loop.

Reply:
Thanks a lot for pointing and the info. I just bluntly followed the SonarCloud suggestion. Changed both `for_each ` to `for`.